conan/tools/google/bazeldeps.py

```diff
@@ -589,3 +645,36 @@ def generate(self):
             bazel_dependencies_module_generator = _BazelDependenciesBZLGenerator(self._conanfile,
                                                                                  deps_info)
             bazel_dependencies_module_generator.generate()
+
+        save(
+            f"{self._conanfile.generators_folder}/conan_deps_repo_rules.bzl",
+            textwrap.dedent("""\
+            # This bazel repository rule is used to load Conan dependencies into the Bazel workspace.
+            # It's used by a generated module file that provides information about the conan packages.
+            # Each conan package is loaded into a bazel repository rule, with having the name of the package.
+            # The whole method is based on symlinks to not copy the whole package into the Bazel workspace,
+            # which is expensive.
+            def _conan_dependency_repo(rctx):
+                package_path = rctx.workspace_root.get_child(rctx.attr.package_path)
+
+                child_packages = package_path.readdir()
+                for child in child_packages:
+                    rctx.symlink(child, child.basename)
+
+                rctx.symlink(rctx.attr.build_file_path, "BUILD.bazel")
```

Comment:
Symlinks always scare me, but it looks like there is no other way to do this.
I need to see how it'd work in more complex scenarios, e.g., using the same dependency as a `tool_require` and a normal requirement (defining the `build_context_activated`).

Reply:
> Symlinks always scare me, but it looks like there is no other way to do this.

I mean there is an alternative, which would be to copy the files. But I'm not sure if this would really be better.
The symlinks point to files in the package folders of Conan packages, those should not be touched under normal conditions, or?
I'm also pretty sure, that the legacy method does the same thing.

> I need to see how it'd work in more complex scenarios, e.g., using the same dependency as a tool_require and a normal requirement (defining the build_context_activated).

Yeah, this is quite interesting. I did not really look into this aspect. But this is anyway the same situation for the legacy Bazel tooling.

Reply:
Ok, thanks a lot for the explanation. Yeah, I think this is a more efficient approach, so let's give it a try with those complex scenarios.

Reply:
symlinks are very standard part of the bazel build infrastructure. I think in this context they are fine.